src/GraphQLParser/Visitors/SDLPrinterExtensions.cs

```diff
@@ -24,13 +24,28 @@ public static string Print(this SDLPrinter printer, ASTNode node)
     public static void Print(this SDLPrinter printer, ASTNode node, StringBuilder stringBuilder)
         => printer.PrintAsync(node, new StringWriter(stringBuilder), default).AsTask().GetAwaiter().GetResult();
 
+#if NET462
+    private static readonly Encoding _uTF8NoBOM = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
+#endif
+
     /// <summary>
     /// Prints the specified AST into the specified <see cref="MemoryStream"/> as a SDL document.
     /// If no encoding is specified, the document is written in UTF-8 format without a byte order mark.
     /// </summary>
     public static void Print(this SDLPrinter printer, ASTNode node, MemoryStream memoryStream, Encoding? encoding = null)
     {
-        using var streamWriter = new StreamWriter(memoryStream, encoding, -1 /* default */, true);
+        int bufferSize = -1;
+#if NET462
```

Comment:
CoreCLR https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/IO/StreamWriter.cs#L94

NET Framework https://github.com/Microsoft/referencesource/blob/master/mscorlib/system/io/streamwriter.cs#L140

Actually this method **does not work at all** in NET Framework for now throwing ArgumentOutOfRangeException. Caller can pass not-null encoding, but buffer size is always negative. 😮

Reply:
I did not touch signatures to not make breaking changes.

Reply:
As option - add new method with bufferSize or change existing arguments in v10 to not bother with additional net462 TFM and all jumping around.

Reply:
Looking at the MS docs, we should not be passing `null` to `encoding`, nor should we be passing `-1` to `bufferSize`.  What we should learn here is that if we do not follow specs, then the code can fail.  For example, what would occur if someone tried to run this on Mono or Xamrin or some other .NET Standard implementation?  We cannot assume that `-1` and `null` would work there any more than it does on .NET Framework.  I would remove the ifdef and leave the code, so that no matter if it is .NET Standard or .NET 6/7/8/9/etc, it will adhere to the current specification, which lists these exceptions:

> ArgumentNullException : path or encoding is null.
> ArgumentOutOfRangeException : buffersize is less than or equal to zero.

We can view the source code of the constructor to determine what the default buffer size would otherwise be.

Reply:
> add new method with bufferSize or change existing arguments in v10

I would not do either at this time.  Our API dictates that null can be passed to encoding, which will use UTF8 with no byte order mark.  Fine.  It doesn't say what buffer size we will use.  Fine.  No breaking changes.  No need to make changes.  Our code simply doesn't work as we documented that it should.

Reply:
Sorry, I stand corrected; we are following the .NET 6 specs; but that overload does not exist for .NET Standard 2.0.  So for .NET 6 it says the arguments are optional when in .NET Standard 2.0/2.1 they are required.  So we can pass `-1` and `null` for .NET 6, but for .NET Standard we must pass correct values.  So let's just use `#if !NET6_0` or whatever rather than adding a TFM to the build process for .NET framework.

Reply:
https://learn.microsoft.com/en-us/dotnet/api/system.io.streamreader.-ctor?view=net-6.0#system-io-streamreader-ctor(system-io-stream-system-text-encoding-system-boolean-system-int32-system-boolean)

https://learn.microsoft.com/en-us/dotnet/api/system.io.streamreader.-ctor?view=netstandard-2.1#system-io-streamreader-ctor(system-io-stream-system-text-encoding-system-boolean-system-int32-system-boolean)

Reply:
Fine. Done.